src/commands/view/summary.ts

```diff
@@ -1,102 +1,154 @@
-      profileSummaries.forEach(profileSummary => {
-        const flattened: Record<string, number> = flat.flatten(profileSummary)
-        Object.entries(flattened).forEach(([key, value]) => {
-          const existingValue = _.get(totals, `${profileName}.${key}`)
-          if (existingValue) {
-            _.set(totals, `["${profileName.replaceAll('"', '\\"')}"].${key}`, existingValue + value)
-          } else {
-            _.set(totals, `["${profileName.replaceAll('"', '\\"')}"].${key}`, value)
-          }
-        })
-      })
-    })
-    const printableSummaries: Record<string, unknown>[] = []
-    Object.entries(totals).forEach(([profileName, profileValues]: any) => {
-      printableSummaries.push({
-        profileName: profileName,
-        // Extract filename from execJSONs
-        resultSets: Object.entries(execJSONs).filter(([, execJSON]) => {
-          return execJSON.data.profiles[0].name === profileName
-        }).map(([filePath]) => {
-          return convertFullPathToFilename(filePath)
-        }),
-        compliance: _.mean(complianceScores[profileName]),
-        ...profileValues,
+    // Enable and disable flags
+    '$ saf summary --input input.hdf --pretty-print                          # Enable human-readable output',
+    '$ saf summary -i input.hdf --no-pretty-print                            # Useful for scripts or data-processing (RAW yaml/json/etc.)',
+  ]
+
+  /**
+   * @private
+   * Holds the parsed command line flags.
+   * @type {CommandFlags}
+   */
+  private parsedFlags!: CommandFlags;
+
+  /**
+   * This is the main function that runs when the 'summary' command is invoked.
+   * It performs the following steps:
+   * 1. Parses the command line flags.
+   * 2. Loads the execution JSONs from the provided input files.
+   * 3. Calculates the summaries for each execution JSON.
+   * 4. Calculates the total counts for each summary.
+   * 5. Calculates the compliance scores for each execution JSON.
+   * 6. Creates a printable summary for each profile using the total counts and compliance scores.
+   * 7. Prints the printable summaries to the console and optionally writes them to an output file.
+   *
+   * @throws {Error} If there's an error during the execution, it will be logged and the process will exit.
+   * @returns {Promise<void>} A promise that resolves when the command has finished executing.
+   */
+  async run() {
+    try {
+      const {flags} = await this.parse(Summary)
+      this.parsedFlags = flags as CommandFlags
+      const {format, 'print-pretty': printPretty, stdout, output, 'title-table': titleTable, logLevel} = flags
+      this.logger = createWinstonLogger(VIEW_SUMMARY, this.parsedFlags.logLevel)
+      this.logger.verbose('Parsed command line flags')
+      const executionData = loadExecJSONs(this.parsedFlags.input)
+      this.logger.verbose(`Loaded execution data from ${this.parsedFlags.input.length} file(s)`)
```

Comment:
```suggestion
      this.logger.verbose(`Loaded execution data from ${this.parsedFlags.input.length.toString()} file(s)`)
```

Copilot suggests this because it would otherwise print [object Object] supposedly.

Reply:
But I didn't check in the logging if this is needed.

Reply:
Please go ahead and make this addition